integration/test/metric_value_benchmark/placeholder_test.go

```diff
@@ -0,0 +1,11 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric_value_benchmark
+
+func (suite *MetricBenchmarkTestSuite) TestDummy() {
```

Comment:
Rather than have this function, would it make more sense to put more context clues in the MetricBenchmarkTestSuite struct? IMO the methodology we should take with this testing "framework" (not sure if you would label this a framework or not but I think it's working towards that) is that it should be clear enough what needs to be done to implement the plumbing for a test. 

Like, we have this currently:
```go
	log.Printf("Testing Cpu values...")
	suite.RunAgent(agentConfigFileName, minimumAgentRuntime)
	testGroupResult := validateCpuMetrics()
	suite.AddToSuiteResult(testGroupResult)
	suite.Assert().Equal(status.SUCCESSFUL, testGroupResult.GetStatus(),
		"Cpu test failed to validate that every metric value is greater than zero")
```
I think we could take one step further by adding something to the test suite definition to encapsulate the real test validation. At a high level, we want to define (per test) some body of work that eventually results in a `status.TestResult`. So we could add something like:
```go
func (s *MetricBenchmarkTestSuite) Validate(f func() status.TestResult) {
    res := f()
    s.Assert().Equal(status.SUCCESSFUL, res)
}
```
and that would cut out the notion of adding the result of the test to a group... which.. maybe you don't want? Idk. Just food for thought. I'll think about it more

Reply:
How about this instead? I considered doing the following and just didn't in order to not go overboard and spend too much time but.. essentially a template method pattern:


```
type TestRunner interface {
   RunTest() 
   runAgent()
   validate() TestGroupResult
   
}
type baseTest struct {
}
func (t *baseTest) RunTest(s* MetricBenchmarkTestSuite) {
   t.runAgent() //to be implemented by each test.. etc cpu test
   t.validate() // to be implemented by each test .. etc cpu test
   s.addToSuiteResult(..) 
}
type CPUTest struct {
}
func (c *CPUTest) runAgent( ){
 .. call shared RunAgent() function with arguments for file path
}
func (c *CPUTest) validate() {
}


var testRunners := []TestRunner // cputest, memtest... etc

func (s *MetricBenchmarkTestSuite) TestSuite() {
    // iterate over all testRunners and call RunTest() on each of them. 
    
    // some assertion for the entire test success/failure
}

```

Reply:
I like it. If we're already trying to refactor this for flexibility, might as well go all out now. I think my one gripe is that everything falls under this one test suite, whereas right now we execute tests individually on a host (e.g.: the metric dimensions test runs on a different host than the cloudwatch logs test). 

With this approach, would we be running multiple tests on the same host? Maybe this points to figuring out what our definition of benchmark is for this. Is it just a sanity test to make sure that we emit **some** value, or are we measuring performance/SLAs/etc.? I'm not _super_ concerned about testing multiple plugins on the same host/test run, but it is something to keep in mind.

Reply:
yeah on the same host. I was thinking maybe we could even run agent once to get most of the metric data and validate values instead of running multiple times. Yeah if it becomes a problem we'll figure it out

Reply:
Alright let me.. have fun and go overboard then haha. Will re-request review after I'm done